src/dmd/backend/dwarfdbginf.d

```diff
@@ -1180,26 +1180,51 @@ static if (1)
         {
             debug_info.initialize();
 
-
-            // Compilation Unit Header
+            void writeCompilationUnitHeader(ubyte hversion)()
             {
-                debug_info.buf.write32(0);                      // unit length
-                debug_info.buf.write16(config.dwarf);           // version
-                if (config.dwarf >= 5)
+                struct CompilationUnitHeader
                 {
-                    debug_info.buf.writeByte(DW_UT_compile);    // Unit Type
-                    debug_info.buf.writeByte(I64 ? 8 : 4);      // Address size
-                    debug_info.buf.write32(0);                  // debug abbrev offset
+                align(1):
+                    uint length;
+                    ushort version_ = hversion;
+                    static if (hversion >= 5)
+                    {
+                        ubyte unit_type = DW_UT_compile;
+                        ubyte address_size = 4;
+                    }
+                    uint debug_abbrev_offset;
+                    static if (hversion < 5)
+                    {
+                        ubyte address_size = 4;
+                    }
                 }
+
+                static if (hversion == 3 || hversion == 4)
+                    static assert(CompilationUnitHeader.sizeof == 11);
+                else static if (hversion == 5)
+                    static assert(CompilationUnitHeader.sizeof == 12);
                 else
+                    static assert(0);
+
+                auto cuh = CompilationUnitHeader.init;
```

Comment:
shouldn't this be initialized with `void` ? Otherwise you are still initializing the fields.

Reply:
Nope, It is written into debug_info at line 1216

Reply:
What I meant is that the compiler is going to allocate space on the stack and initialize that space with `CompilationUnitHeader.init`. Whereas if void initialization is used [1], the initialization phase is skipped.

[1] https://dlang.org/spec/declaration.html#void_init

Reply:
I want it to be initialized, otherwise bad values will be written.

Quick example: https://run.dlang.io/is/sQV78B